src/lib/tls/tls_handshake_io.h

```diff
@@ -177,8 +177,6 @@ class Datagram_Handshake_IO final : public Handshake_IO
          Message_Info(uint16_t e, Handshake_Type mt, const std::vector<uint8_t>& msg) :
             epoch(e), msg_type(mt), msg_bits(msg) {}
 
-         Message_Info(const Message_Info& other) = default;
```

Comment:
No, `Message_Info` is just a struct used internally by (and is private to) `Datagram_Handshake_IO`, used for saving message flights for retransmission. It should be `final`, as should `Handshake_Reassembly` in the same header.